app/src/main/java/net/grandcentrix/blutufintegration/ui/list/ListViewModel.kt

```diff
@@ -41,15 +48,19 @@ class ListViewModel : ViewModel() {
         }
     }
 
-    @ExperimentalCoroutinesApi
     fun startScan() {
         viewModelScope.launch {
+            scanState.value = true
             BluetoothRepository.scan()
+            Handler(Looper.getMainLooper()).postDelayed({
```

Comment:
This is a way to stop the scan after some time. But this could be done also with Kotlin Coroutines here. Keyword `delay()`. In my view this is a bit more elegant and easy.